text/0086-sentry-bundler-plugins-api.md

```diff
@@ -0,0 +1,110 @@
+  sourcemaps?: {
+    assets: string | string[]; // globs pointing to the javascript files that should be uploaded to Sentry with their source maps - these are the built assets and not the source files
+    ignore?: string | string[]; // globs to exclude javascript files from being uploaded (will also not upload their source maps)
+    rewriteSources?: (source: string) => string; // Hook to rewrite individual entries in the sourcemaps's `sources` field. By default, if not defined, the plugin will try to rewrite the entries to be relative to `process.cwd()`.
+  };
+
+  // General configuration for release creation and release injection - we decouple this from the debug ID source maps upload because they're technically not related anymore.
+  // If not provided, this option will default to its default values.
+  release?: {
+    name?: string; // if not provided - Sentry CLI will try to guess a name (ie. git sha, cloud provider env vars, etc.)
+    inject?: boolean; // default: true - whether to inject the release into bundles
+    create?: boolean; // default: true - if set to `false` will not create a release in Sentry, no matter what other options are set - the release value will still be injected though.
+    finalize?: boolean; // default: true
+    dist?: string; // Can be used to segment releases further
+    vcsRemote?: string; // default: 'origin' - Version control system remote name
+    setCommits?: SetCommitsOptions;
+    deploy?: DeployOptions;
+    legacySourcemaps?: {
+      // Pre-debugid way of uploading source maps - for users that for whatever reason cannot inject code into their bundles
+      include: string | IncludeEntry | Array<string | IncludeEntry>;
+      cleanArtifacts?: boolean; // default: false - Wether to delete artifacts previously uploaded to the release
+    };
+  };
+
+  _experiments: {}; // whatever
+}
+
+// Identical to status-quo. See https://github.com/getsentry/sentry-webpack-plugin/blob/2b7d274a7355f0d27a431b2c20c37c9786bbe4cb/README.md for more information.
+interface IncludeEntry {
+  // ...
+}
+
+// Identical to status-quo. See https://github.com/getsentry/sentry-webpack-plugin/blob/2b7d274a7355f0d27a431b2c20c37c9786bbe4cb/README.md for more information.
+interface SetCommitsOptions {
+  // ...
+}
+```
+
+Some removed options:
+
+- `dryRun` - Current assumption is that this option is unnecessary:
+  - Debug information can be printed with the `debug` option instead.
+  - Executing the plugin is **very** low stakes. Releases and artifact bundles can be deleted with ease and without any lasting side-effects.
+  - This option comes with maintenance effort that scales linearly with new features and there is a high chance implementing is overlooked for new features.
+  - We introduce a `disabled` option instead that still is a convenient way of disabling the plugin without any continuous maintenance effort.
+- `configFile`
```

Comment:
+1! Based on my understanding, removing this option would anyways still allow the CLI to access a `.sentryclirc` file if one is present.

Reply:
I guess so. In my "perfect world" it wouldn't even pick up that file but whatever. If users discover the `.sentryclirc` feature they have gone so deep they deserve to use it.